src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -1109,6 +1091,38 @@ public ChannelMigrateOffsetTokenResponseDTO migrateStreamingChannelOffsetToken(
     }
   }
 
+  @Override
+  public Optional<List<DescribeTableRow>> describeTable(String tableName) {
+    checkConnection();
+    String query = "desc table identifier(?)";
+    PreparedStatement stmt = null;
+    List<DescribeTableRow> rows = new ArrayList<>();
+
+    try {
+      stmt = conn.prepareStatement(query);
+      stmt.setString(1, tableName);
+      ResultSet result = stmt.executeQuery();
+
+      while (result.next()) {
+        String columnName = result.getString(1);
```

Comment:
we could also use names of the columns, bit safer but i guess the order won;t change in the future
```suggestion
        String columnName = result.getString("name");
```

Reply:
Good point, it is not likely but might break the whole connector some day.